parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -398,6 +409,14 @@ pub mod pallet {
 			)?;
 
 			let current_period = Self::compute_current_sync_period(update.attested_header.slot);
+			let latest_committee_period = <LatestSyncCommitteePeriod<T>>::get();
+			let next_period = current_period + 1;
+			ensure!(
+				(next_period == latest_committee_period + 1) &&
+					!<SyncCommittees<T>>::contains_key(next_period),
```

Comment:
I think the `!<SyncCommittees<T>>::contains_key(next_period)` check should perhaps be separate and fail with `SyncCommitteeMissing`, because the sync committee update could be valid but the sync committee for that period has not been imported yet.

Reply:
Hey Clara. I think maybe it is not a `SyncCommitteeMissing` and my intention for `!<SyncCommittees<T>>::contains_key(next_period)` here is to ensure we not import a duplicated `SyncCommittees`

And I did some [refactoring](https://github.com/Snowfork/snowbridge/pull/773/commits/364c05c1896c68ad1fb257245ff529a817260234) here to seperate the error code a bit clear as you suggest